RecoLocalCalo/HcalRecAlgos/src/HFStripFilter.cc

```diff
@@ -0,0 +1,361 @@
+       << ")  Lstrip2 = " << (int)d2strip.size() << std::endl << " Strip1: ";
+    for (auto& it1 : d1strip) {
+      ss << it1.energy() << " (" << it1.id().ieta() << ") "; }
+    ss << std::endl << " Strip2: ";
+    for (auto& it1 : d2strip) {
+      ss << it1.energy() << " (" << it1.id().ieta() << ") "; }
+    ss << std::endl;
+  }
+  
+  // check if one of strips in depth1 or depth2 >= lstrips_
+  if ((int)d1strip.size() < lstrips_ && (int)d2strip.size() < lstrips_) {
+    if (verboseLevel_ >= 30) {
+      LogDebug("HFSFilter") << ss.str();
+    }
+    return;
+  } 
+  
+  // define range of strips in ieta 
+  int ietaMin1 = 1000;  // for d1strip
+  int ietaMax1 = -1000;
+  for (auto& it1 : d1strip) {
+    if (it1.id().ieta() < ietaMin1) ietaMin1 = it1.id().ieta();
+    if (it1.id().ieta() > ietaMax1) ietaMax1 = it1.id().ieta();
+  }
+  int ietaMin2 = 1000;  // for d2strip
+  int ietaMax2 = -1000;
+  for (auto& it1 : d2strip) {
+    if (it1.id().ieta() < ietaMin2) ietaMin2 = it1.id().ieta();
+    if (it1.id().ieta() > ietaMax2) ietaMax2 = it1.id().ieta();
+  }
+  
+  // define ietamin and ietamax - common area for d1strip and d2strip
+  int ietaMin = ietaMin1;
+  int ietaMax = ietaMax1;
+  
+  if (ietaMin2 >= ietaMin1 && ietaMin2 <= ietaMax1) {
+    if (ietaMax2 > ietaMax1) ietaMax = ietaMax2;
+  } else if (ietaMin2 <= ietaMin1 && ietaMax2 >= ietaMin1) {
+    if (ietaMin2 < ietaMin1) ietaMin = ietaMin2;
+    if (ietaMax2 > ietaMax1) ietaMax = ietaMax2;
+  }
+    
+  // calculate the total energy in strips
+  double eStrip = 0;    
+  for (auto& it1 : d1strip) {   
+    if (it1.id().ieta() < ietaMin || it1.id().ieta() > ietaMax) continue;
+    eStrip += it1.energy();
+  }
+  for (auto& it1 : d2strip) {   
+    if (it1.id().ieta() < ietaMin || it1.id().ieta() > ietaMax) continue;
+    eStrip += it1.energy();
+  }
+  
+  if (verboseLevel_ >= 30) {
+    ss << " ietaMin1 = " << ietaMin1 << "  ietaMax1 = " << ietaMax1 << std::endl
+       << " ietaMin2 = " << ietaMin2 << "  ietaMax2 = " << ietaMax2 << std::endl   
+       << " Common strip:  ietaMin = " << ietaMin << "  ietaMax = " << ietaMax << std::endl; 
+  }
+  
+  // energies in the neighboring wedges  
+  double energyIphi1 = 0;
+  double energyIphi2 = 0;
+  int iphi1 = 0, iphi2 = 0;
+
+  // get information about the neighboring wedges from HcalTopology
+  HcalDetId neighbour;
+  
+  for (auto& it : rec)
+    {
+      if (it.energy() < stripThreshold_) continue;
+      if (it.id().ieta() < ietaMin || it.id().ieta() > ietaMax) continue;
+      HcalDetId id(HcalForward, it.id().ieta(), stripIphiMax, stripDepthMax);
+      bool neigh = myqual->topo()->decIPhi(id, neighbour);
+      iphi1 = (neigh) ? neighbour.iphi() : 0;
+      neigh = myqual->topo()->incIPhi(id, neighbour);
+      iphi2 = (neigh) ? neighbour.iphi() : 0;
+      if (it.id().iphi() == iphi1) energyIphi1 += it.energy();      // Energy iphi1
+      else if (it.id().iphi() == iphi2) energyIphi2 += it.energy(); // Energy iphi2
```

Comment:
(Just to check the logic): is it really what you want to add up the energy in only one of the two neighbours?